--- ../mdb/mdb.c.orig	2016-02-28 21:18:07.000000000 +0300
+++ ../mdb/mdb.c	2016-02-28 21:15:28.000000000 +0300
@@ -6896,7 +6896,7 @@
 mdb_cursor_get(MDB_cursor *mc, MDB_val *key, MDB_val *data,
     MDB_cursor_op op)
 {
-	int		 rc;
+	int		 rc = 0;
 	int		 exact = 0;
 	int		 (*mfunc)(MDB_cursor *mc, MDB_val *key, MDB_val *data);
 
@@ -10692,9 +10692,11 @@
 	for (i=0; i<rdrs; i++) {
 		if (mr[i].mr_pid) {
 			txnid_t	txnid = mr[i].mr_txnid;
-			sprintf(buf, txnid == (txnid_t)-1 ?
-				"%10d %"Z"x -\n" : "%10d %"Z"x %"Y"u\n",
-				(int)mr[i].mr_pid, (size_t)mr[i].mr_tid, txnid);
+			if (txnid == (txnid_t)-1) {
+				sprintf(buf, "%10d %"Z"x -\n", (int)mr[i].mr_pid, (size_t)mr[i].mr_tid);
+			} else {
+				sprintf(buf, "%10d %"Z"x %"Z"u\n", (int)mr[i].mr_pid, (size_t)mr[i].mr_tid, txnid);
+			}
 			if (first) {
 				first = 0;
 				rc = func("    pid     thread     txnid\n", ctx);
